widgetry/src/backend_glow_native.rs

```diff
@@ -97,3 +107,29 @@ impl WindowAdapter {
         self.0.swap_buffers().unwrap();
     }
 }
+
+unsafe fn compile_shaders(
+    gl: &glow::Context,
+    vertex_source: &str,
+    fragment_source: &str,
+) -> Result<[u32; 2], Box<dyn std::error::Error>> {
```

Comment:
Nit: I think `Vec<u32>` is fine for the successful result. An array might be slightly more performant, but it's less readable at the callsite to have to say `.iter()`. The heap allocation should be negligible (and maybe `vec![]` does something clever anyway)

Reply:
When something has a known size, I prefer to keep that info in the type system in case it's handy - it wasn't about squeezing perf out of a u32 Vec of length 2 in code that's called once. =)

I've gotten rid of the unnecessary `iter` to make the call sight clearer.

But also if you just really like `Vec`s for whatever reason I can accommodate that.

Reply:
Ah, I misinterpreted the intention behind the array. More explicit types are a good idea.